pkg/jtracer/jtracer.go

```diff
@@ -14,16 +14,57 @@
 
 package jtracer
 
-import "github.com/opentracing/opentracing-go"
+import (
+	"context"
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/propagation"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
 
 type JTracer struct {
-	OT opentracing.Tracer
+	OT   opentracing.Tracer
+	OTEL trace.TracerProvider
 }
 
 func OT(t opentracing.Tracer) JTracer {
 	return JTracer{OT: t}
 }
 
+func OTEL(tp trace.TracerProvider) JTracer {
+	return JTracer{OTEL: tp}
+}
+
 func NoOp() JTracer {
 	return JTracer{OT: opentracing.NoopTracer{}}
 }
+
+func NoOpOtel() JTracer {
+	return JTracer{OTEL: trace.NewNoopTracerProvider()}
+}
+
+func NewOTELProvider() (trace.TracerProvider, error) {
+	opts := []otlptracehttp.Option{otlptracehttp.WithInsecure()}
+	exporter, err := otlptrace.New(
+		context.Background(),
+		otlptracehttp.NewClient(opts...),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create exporter: %w", err)
+	}
+
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exporter),
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
```

Comment:
two considerations here
  * we should allow the exact settings to be customizable, e.g. by relying on OTEL's env vars
  * we should also have some sensible defaults when no env vars are given